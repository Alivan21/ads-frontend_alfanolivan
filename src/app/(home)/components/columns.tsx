"use client";

import Link from "next/link";
import { Button } from "@/components/ui/button";
import { Book } from "@/types/book";
import { formatPrice } from "@/utils/format-idr";
import { ColumnDef } from "@tanstack/react-table";
import { ArrowUpDown, EyeIcon, Pencil, Trash2 } from "lucide-react";
import DeleteModal from "./delete-modal";

export const columns: ColumnDef<Book>[] = [
  {
    accessorKey: "title",
    header: () => {
      return <span className="whitespace-nowrap uppercase">Judul</span>;
    },
    cell: ({ row }) => {
      return <span className="whitespace-nowrap">{row.getValue("title")}</span>;
    },
  },
  {
    accessorKey: "description",
    header: "Deskripsi",
    cell: ({ row }) => {
      return <span className="line-clamp-2 max-w-md whitespace-nowrap">{row.getValue("description")}</span>;
    },
  },
  {
    accessorKey: "price",
    header: ({ column }) => {
      return (
        <Button onClick={() => column.toggleSorting(column.getIsSorted() === "asc")} variant="ghost">
          <span className="uppercase">Harga</span>
          <ArrowUpDown className="ml-2 h-4 w-4" />
        </Button>
      );
    },
    cell: ({ row }) => {
      const formatted = formatPrice(row.getValue("price"));
      return <span>{formatted}</span>;
    },
  },
  {
    accessorKey: "author",
    header: "Penulis",
  },
  {
    id: "actions",
    header: "Aksi",
    cell: ({ row }) => {
      const book = row.original;
      return (
        <div className="flex flex-col gap-2 md:flex-row">
          <Link href={`/detail/${book.id}`}>
            <Button className="bg-blue-600 hover:bg-blue-700" size="sm">
              <EyeIcon size={20} strokeWidth={2.5} />
            </Button>
          </Link>
          <Link href={`/edit/${book.id}`}>
            <Button className="bg-green-600 hover:bg-green-700" size="sm">
              <Pencil size={20} strokeWidth={2.5} />
            </Button>
          </Link>
          <DeleteModal id={book.id} />
        </div>
      );
    },
  },
];
