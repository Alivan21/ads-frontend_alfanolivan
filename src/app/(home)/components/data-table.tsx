"use client";

import * as React from "react";
import Link from "next/link";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from "@/components/ui/table";
import {
  ColumnDef,
  ColumnFiltersState,
  flexRender,
  getCoreRowModel,
  getFilteredRowModel,
  getPaginationRowModel,
  getSortedRowModel,
  SortingState,
  useReactTable,
} from "@tanstack/react-table";
import { ChevronLeft, ChevronRight, Plus } from "lucide-react";

interface DataTableProps<TData, TValue> {
  columns: ColumnDef<TData, TValue>[];
  data: TData[];
}

export function DataTable<TData, TValue>({ columns, data }: DataTableProps<TData, TValue>) {
  const [sorting, setSorting] = React.useState<SortingState>([]);
  const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>([]);

  const table = useReactTable({
    data,
    columns,
    getCoreRowModel: getCoreRowModel(),
    getPaginationRowModel: getPaginationRowModel(),
    onSortingChange: setSorting,
    getSortedRowModel: getSortedRowModel(),
    onColumnFiltersChange: setColumnFilters,
    getFilteredRowModel: getFilteredRowModel(),
    state: {
      sorting,
      columnFilters,
    },
  });

  return (
    <>
      <div className="flex flex-col gap-2 md:flex-row md:items-center md:justify-between md:gap-0">
        <Link href="/create">
          <Button className="flex items-center gap-1.5 bg-blue-600 hover:bg-blue-700">
            <Plus size={20} strokeWidth={2.5} /> Tambah Buku
          </Button>
        </Link>
        <Input
          className="max-w-sm"
          onChange={event => table.getColumn("title")?.setFilterValue(event.target.value)}
          placeholder="Filter Judul..."
          value={(table.getColumn("title")?.getFilterValue() as string) ?? ""}
        />
      </div>
      <Table>
        <TableHeader className="bg-blue-600 uppercase">
          {table.getHeaderGroups().map(headerGroup => (
            <TableRow key={headerGroup.id}>
              {headerGroup.headers.map(header => {
                return (
                  <TableHead className="text-white" key={header.id}>
                    {header.isPlaceholder ? null : flexRender(header.column.columnDef.header, header.getContext())}
                  </TableHead>
                );
              })}
            </TableRow>
          ))}
        </TableHeader>
        <TableBody className="rounded-md border">
          {table.getRowModel().rows?.length ? (
            table.getRowModel().rows.map(row => (
              <TableRow className="hover:bg-muted/50" data-state={row.getIsSelected() ? "selected" : null} key={row.id}>
                {row.getVisibleCells().map(cell => (
                  <TableCell key={cell.id}>{flexRender(cell.column.columnDef.cell, cell.getContext())}</TableCell>
                ))}
              </TableRow>
            ))
          ) : (
            <TableRow className="hover:bg-muted/50">
              <TableCell className="h-24 text-center" colSpan={columns.length}>
                No results.
              </TableCell>
            </TableRow>
          )}
        </TableBody>
      </Table>
      <div className="flex flex-col gap-2 px-3 md:flex-row md:items-center md:justify-between">
        <div className="flex flex-row items-center justify-between md:justify-start md:gap-5">
          <div>
            <span>Page</span>
            <strong>
              {" "}
              {table.getState().pagination.pageIndex + 1} of {table.getPageCount()}
            </strong>
          </div>
          <div className="flex items-center gap-1">
            Row Per Page:{" "}
            <select
              className="rounded-md border p-1"
              name="perpage"
              onChange={e => table.setPageSize(parseInt(e.target.value))}
              value={table.getState().pagination.pageSize}
            >
              {[10, 20, 30, 40, 50].map(pageSize => (
                <option key={pageSize} value={pageSize}>
                  {pageSize}
                </option>
              ))}
            </select>
          </div>
        </div>
        <div className="flex items-center justify-between md:justify-end md:space-x-2">
          <Button
            disabled={!table.getCanPreviousPage()}
            onClick={() => table.previousPage()}
            size="sm"
            variant="outline"
          >
            <ChevronLeft />
          </Button>
          <Button disabled={!table.getCanNextPage()} onClick={() => table.nextPage()} size="sm" variant="outline">
            <ChevronRight />
          </Button>
        </div>
      </div>
    </>
  );
}
