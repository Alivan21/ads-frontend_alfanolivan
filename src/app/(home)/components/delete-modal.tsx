"use client";

import { Button } from "@/components/ui/button";
import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogFooter,
  DialogHeader,
  DialogTitle,
  DialogTrigger,
} from "@/components/ui/dialog";
import { DialogClose } from "@radix-ui/react-dialog";
import { Trash2 } from "lucide-react";
import { useDeleteBook } from "../services/delete-book";

function DeleteModal({ id }: { id: number }) {
  const { mutateAsync: deleteBook } = useDeleteBook();

  const handleDelete = async () => {
    await deleteBook(id);
  };

  return (
    <Dialog>
      <DialogTrigger asChild>
        <Button size="sm" variant="destructive">
          <Trash2 />
        </Button>
      </DialogTrigger>
      <DialogContent className="sm:max-w-[425px]">
        <DialogHeader>
          <DialogTitle>Hapus Buku</DialogTitle>
          <DialogDescription>
            Apakah anda yakin ingin menghapus buku ini? Buku yang sudah dihapus tidak dapat dikembalikan.
          </DialogDescription>
        </DialogHeader>
        <DialogFooter>
          <DialogClose asChild>
            <Button onClick={handleDelete} variant="destructive">
              Hapus
            </Button>
          </DialogClose>
          <DialogClose asChild>
            <Button type="button" variant="outline">
              Kembali
            </Button>
          </DialogClose>
        </DialogFooter>
      </DialogContent>
    </Dialog>
  );
}
export default DeleteModal;
