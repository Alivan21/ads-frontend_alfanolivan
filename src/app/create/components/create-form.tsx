"use client";

import { ChangeEvent, useState } from "react";
import { useRouter } from "next/navigation";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Textarea } from "@/components/ui/textarea";
import { Book } from "@/types/book";
import toast from "react-hot-toast";
import { useCreateBook } from "../services/create-book";

type BookForm = Omit<Book, "id">;

function CreateForm() {
  const router = useRouter();
  const [form, setForm] = useState<BookForm>({
    title: "",
    author: "",
    price: "" as unknown as number,
    description: "",
  });
  const { isPending, mutateAsync: createBook } = useCreateBook(form);

  const handleChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
    const { id, value } = event.target;
    setForm(prevForm => ({
      ...prevForm,
      [id]: value,
    }));
  };

  const handleInput = (e: ChangeEvent<HTMLInputElement>) => {
    e.target.value = e.target.value.replace(/[^0-9]/g, "");
  };

  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    try {
      await createBook();
      router.push("/");
    } catch (error) {
      toast.error("Gagal menambahkan buku");
    }
  };

  return (
    <form className="flex flex-col gap-4" onSubmit={handleSubmit}>
      <div className="grid grid-cols-2 gap-4">
        <div className="flex flex-col">
          <label className="mb-2 font-medium" htmlFor="title">
            Judul
          </label>
          <Input id="title" onChange={handleChange} placeholder="Masukkan judul" value={form.title} />
        </div>
        <div className="flex flex-col">
          <label className="mb-2 font-medium" htmlFor="author">
            Penulis
          </label>
          <Input id="author" onChange={handleChange} placeholder="Masukkan nama pengarang" value={form.author} />
        </div>
      </div>
      <div className="flex flex-col">
        <label className="mb-2 font-medium" htmlFor="price">
          Harga
        </label>
        <Input
          id="price"
          onChange={handleChange}
          onInput={handleInput}
          pattern="[0-9]*"
          placeholder="Masukkan harga"
          type="text"
          value={form.price}
        />
      </div>
      <div className="mb-2 flex flex-col">
        <label className="mb-2 font-medium" htmlFor="description">
          Deskripsi
        </label>
        <Textarea
          id="description"
          onChange={handleChange}
          placeholder="Ketik deskripsi disini"
          value={form.description}
        />
      </div>
      <Button className="w-full bg-blue-600 hover:bg-blue-700" disabled={isPending} type="submit">
        {isPending ? "Menambahkan..." : "Tambah"}
      </Button>
    </form>
  );
}
export default CreateForm;
