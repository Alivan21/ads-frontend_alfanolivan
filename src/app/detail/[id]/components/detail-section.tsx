"use client";

import Spinner from "@/components/spinner";
import { formatPrice } from "@/utils/format-idr";
import { useQuery } from "@tanstack/react-query";
import { getBookById } from "../services/book-by-id";

function DetailSection({ id }: { id: string }) {
  const { data, isLoading } = useQuery({
    queryKey: ["book", id],
    queryFn: () => getBookById(id),
  });

  if (isLoading || !data) {
    return (
      <div className="flex items-center justify-center p-8">
        <Spinner />
      </div>
    );
  }

  return (
    <div className="space-y-4">
      <div>
        <h2 className="text-lg font-semibold">Judul</h2>
        <p>{data.data.title}</p>
      </div>
      <div>
        <h2 className="text-lg font-semibold">Deskripsi</h2>
        <p>{data.data.description}</p>
      </div>
      <div>
        <h2 className="text-lg font-semibold">Penulis</h2>
        <p>{data.data.author}</p>
      </div>
      <div>
        <h2 className="text-lg font-semibold">Harga</h2>
        <p>{formatPrice(data.data.price)}</p>
      </div>
    </div>
  );
}
export default DetailSection;
