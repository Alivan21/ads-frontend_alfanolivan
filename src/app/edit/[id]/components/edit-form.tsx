"use client";

import { ChangeEvent, useEffect, useState } from "react";
import { useRouter } from "next/navigation";
import Spinner from "@/components/spinner";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Textarea } from "@/components/ui/textarea";
import { Book } from "@/types/book";
import toast from "react-hot-toast";
import { useGetBookById } from "../services/book-by-id";
import { useUpdateBook } from "../services/update-book";

function EditForm({ id }: { id: string }) {
  const router = useRouter();
  const { data, isLoading } = useGetBookById(id);
  const [form, setForm] = useState<Book>({
    id: "" as unknown as number,
    title: "",
    author: "",
    price: "" as unknown as number,
    description: "",
  });

  useEffect(() => {
    if (data) {
      setForm(data.data);
    }
  }, [data]);

  const handleChange = (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
    const { id, value } = event.target;
    setForm(prevForm => ({
      ...prevForm,
      [id]: value,
    }));
  };

  const handleInput = (e: ChangeEvent<HTMLInputElement>) => {
    e.target.value = e.target.value.replace(/[^0-9]/g, "");
  };

  const { isPending, mutateAsync: updateBook } = useUpdateBook(form);

  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    try {
      await updateBook();
      router.push("/");
    } catch (error) {
      toast.error("Gagal mengubah buku");
    }
  };

  if (isLoading || !data) {
    return (
      <div className="flex items-center justify-center p-8">
        <Spinner />
      </div>
    );
  }

  return (
    <form className="flex flex-col gap-4" onSubmit={handleSubmit}>
      <div className="grid grid-cols-2 gap-4">
        <div className="flex flex-col">
          <label className="mb-2 font-medium" htmlFor="title">
            Judul
          </label>
          <Input id="title" onChange={handleChange} placeholder="Masukkan judul" value={form.title} />
        </div>
        <div className="flex flex-col">
          <label className="mb-2 font-medium" htmlFor="author">
            Penulis
          </label>
          <Input id="author" onChange={handleChange} placeholder="Masukkan nama pengarang" value={form.author} />
        </div>
      </div>
      <div className="flex flex-col">
        <label className="mb-2 font-medium" htmlFor="price">
          Harga
        </label>
        <Input
          id="price"
          onChange={handleChange}
          onInput={handleInput}
          pattern="[0-9]*"
          placeholder="Masukkan harga"
          type="text"
          value={form.price}
        />
      </div>
      <div className="mb-2 flex flex-col">
        <label className="mb-2 font-medium" htmlFor="description">
          Deskripsi
        </label>
        <Textarea
          id="description"
          onChange={handleChange}
          placeholder="Ketik deskripsi disini"
          value={form.description}
        />
      </div>
      <Button className="w-full bg-blue-600 hover:bg-blue-700" disabled={isPending} type="submit">
        {isPending ? "Mengubah..." : "Ubah"}
      </Button>
    </form>
  );
}

export default EditForm;
