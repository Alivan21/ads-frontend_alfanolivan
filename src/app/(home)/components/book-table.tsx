"use client";

import Spinner from "@/components/spinner";
import { useQuery } from "@tanstack/react-query";
import { getAllBook } from "../services/all-book";
import { columns } from "./columns";
import { DataTable } from "./data-table";

function BookTable() {
  const { data, isLoading } = useQuery({
    queryKey: ["books"],
    queryFn: getAllBook,
  });

  if (isLoading || !data) {
    return <Spinner />;
  }

  return <DataTable columns={columns} data={data?.data} />;
}
export default BookTable;
