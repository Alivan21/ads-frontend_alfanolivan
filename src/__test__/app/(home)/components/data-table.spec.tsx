import { columns } from "@/app/(home)/components/columns";
import { DataTable } from "@/app/(home)/components/data-table";
import { render } from "@testing-library/react";

describe("Data Table Component", () => {
  it("should render data table component", () => {
    const page = render(<DataTable columns={columns} data={[]} />);
    expect(page).toMatchSnapshot();
  });
});
