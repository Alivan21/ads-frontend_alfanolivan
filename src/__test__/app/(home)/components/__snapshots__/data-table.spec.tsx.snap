// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Component should render data table component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-2 md:flex-row md:items-center md:justify-between md:gap-0"
      >
        <a
          href="/create"
        >
          <button
            class="justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary-foreground h-10 px-4 py-2 flex items-center gap-1.5 bg-blue-600 hover:bg-blue-700"
          >
            <svg
              class="lucide lucide-plus "
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2.5"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 12h14"
              />
              <path
                d="M12 5v14"
              />
            </svg>
             Tambah Buku
          </button>
        </a>
        <div
          class="relative w-full max-w-sm"
        >
          <svg
            class="lucide lucide-search absolute left-2.5 top-3 h-4 w-4 text-gray-500"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <path
              d="m21 21-4.3-4.3"
            />
          </svg>
          <input
            class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 pl-9"
            placeholder="Filter Judul..."
            value=""
          />
        </div>
      </div>
      <div
        class="relative w-full overflow-auto"
      >
        <table
          class="w-full caption-bottom text-sm"
        >
          <thead
            class="[&_tr]:border-b bg-blue-600 uppercase"
          >
            <tr
              class="border-b transition-colors data-[state=selected]:bg-muted"
            >
              <th
                class="h-12 px-4 text-left align-middle font-medium [&:has([role=checkbox])]:pr-0 text-white"
              >
                <span
                  class="whitespace-nowrap uppercase"
                >
                  Judul
                </span>
              </th>
              <th
                class="h-12 px-4 text-left align-middle font-medium [&:has([role=checkbox])]:pr-0 text-white"
              >
                Deskripsi
              </th>
              <th
                class="h-12 px-4 text-left align-middle font-medium [&:has([role=checkbox])]:pr-0 text-white"
              >
                <button
                  class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
                >
                  <span
                    class="uppercase"
                  >
                    Harga
                  </span>
                  <svg
                    class="lucide lucide-arrow-up-down ml-2 h-4 w-4"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m21 16-4 4-4-4"
                    />
                    <path
                      d="M17 20V4"
                    />
                    <path
                      d="m3 8 4-4 4 4"
                    />
                    <path
                      d="M7 4v16"
                    />
                  </svg>
                </button>
              </th>
              <th
                class="h-12 px-4 text-left align-middle font-medium [&:has([role=checkbox])]:pr-0 text-white"
              >
                Penulis
              </th>
              <th
                class="h-12 px-4 text-left align-middle font-medium [&:has([role=checkbox])]:pr-0 text-white"
              >
                Aksi
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0 rounded-md border"
          >
            <tr
              class="border-b transition-colors data-[state=selected]:bg-muted hover:bg-muted/50"
            >
              <td
                class="p-4 align-middle [&:has([role=checkbox])]:pr-0 h-24 text-center"
                colspan="5"
              >
                No results.
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="flex flex-col gap-2 px-3 md:flex-row md:items-center md:justify-between"
      >
        <div
          class="flex flex-row items-center justify-between md:justify-start md:gap-5"
        >
          <div>
            <span>
              Page
            </span>
            <strong>
               
              1
               of 
              0
            </strong>
          </div>
          <div
            class="flex items-center gap-1"
          >
            Row Per Page:
             
            <select
              class="rounded-md border p-1"
              name="perpage"
            >
              <option
                value="10"
              >
                10
              </option>
              <option
                value="20"
              >
                20
              </option>
              <option
                value="30"
              >
                30
              </option>
              <option
                value="40"
              >
                40
              </option>
              <option
                value="50"
              >
                50
              </option>
            </select>
          </div>
        </div>
        <div
          class="flex items-center justify-between md:justify-end md:space-x-2"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
            disabled=""
          >
            <svg
              class="lucide lucide-chevron-left "
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m15 18-6-6 6-6"
              />
            </svg>
          </button>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
            disabled=""
          >
            <svg
              class="lucide lucide-chevron-right "
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-2 md:flex-row md:items-center md:justify-between md:gap-0"
    >
      <a
        href="/create"
      >
        <button
          class="justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary-foreground h-10 px-4 py-2 flex items-center gap-1.5 bg-blue-600 hover:bg-blue-700"
        >
          <svg
            class="lucide lucide-plus "
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2.5"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 12h14"
            />
            <path
              d="M12 5v14"
            />
          </svg>
           Tambah Buku
        </button>
      </a>
      <div
        class="relative w-full max-w-sm"
      >
        <svg
          class="lucide lucide-search absolute left-2.5 top-3 h-4 w-4 text-gray-500"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <path
            d="m21 21-4.3-4.3"
          />
        </svg>
        <input
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 pl-9"
          placeholder="Filter Judul..."
          value=""
        />
      </div>
    </div>
    <div
      class="relative w-full overflow-auto"
    >
      <table
        class="w-full caption-bottom text-sm"
      >
        <thead
          class="[&_tr]:border-b bg-blue-600 uppercase"
        >
          <tr
            class="border-b transition-colors data-[state=selected]:bg-muted"
          >
            <th
              class="h-12 px-4 text-left align-middle font-medium [&:has([role=checkbox])]:pr-0 text-white"
            >
              <span
                class="whitespace-nowrap uppercase"
              >
                Judul
              </span>
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium [&:has([role=checkbox])]:pr-0 text-white"
            >
              Deskripsi
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium [&:has([role=checkbox])]:pr-0 text-white"
            >
              <button
                class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
              >
                <span
                  class="uppercase"
                >
                  Harga
                </span>
                <svg
                  class="lucide lucide-arrow-up-down ml-2 h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m21 16-4 4-4-4"
                  />
                  <path
                    d="M17 20V4"
                  />
                  <path
                    d="m3 8 4-4 4 4"
                  />
                  <path
                    d="M7 4v16"
                  />
                </svg>
              </button>
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium [&:has([role=checkbox])]:pr-0 text-white"
            >
              Penulis
            </th>
            <th
              class="h-12 px-4 text-left align-middle font-medium [&:has([role=checkbox])]:pr-0 text-white"
            >
              Aksi
            </th>
          </tr>
        </thead>
        <tbody
          class="[&_tr:last-child]:border-0 rounded-md border"
        >
          <tr
            class="border-b transition-colors data-[state=selected]:bg-muted hover:bg-muted/50"
          >
            <td
              class="p-4 align-middle [&:has([role=checkbox])]:pr-0 h-24 text-center"
              colspan="5"
            >
              No results.
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="flex flex-col gap-2 px-3 md:flex-row md:items-center md:justify-between"
    >
      <div
        class="flex flex-row items-center justify-between md:justify-start md:gap-5"
      >
        <div>
          <span>
            Page
          </span>
          <strong>
             
            1
             of 
            0
          </strong>
        </div>
        <div
          class="flex items-center gap-1"
        >
          Row Per Page:
           
          <select
            class="rounded-md border p-1"
            name="perpage"
          >
            <option
              value="10"
            >
              10
            </option>
            <option
              value="20"
            >
              20
            </option>
            <option
              value="30"
            >
              30
            </option>
            <option
              value="40"
            >
              40
            </option>
            <option
              value="50"
            >
              50
            </option>
          </select>
        </div>
      </div>
      <div
        class="flex items-center justify-between md:justify-end md:space-x-2"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
          disabled=""
        >
          <svg
            class="lucide lucide-chevron-left "
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m15 18-6-6 6-6"
            />
          </svg>
        </button>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
          disabled=""
        >
          <svg
            class="lucide lucide-chevron-right "
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
